### ISSUE TITLE
YamlConfig resources cannot be deleted

### ISSUE THREAD

**Issue by eiladin** [harness-io/terraform-provider-harness#46]
The YamlConfig resource cannot be deleted.  The code for the provider is using the wrong client and method.  It's attempting to delete a CloudProvider instead of using the ConfigAsCodeClient to remove the yaml document.

 https://github.com/harness-io/terraform-provider-harness/blob/main/internal/service/cd/yamlconfig/yaml_config.go#L121

### Terraform Version
1.0.11

### Affected Resource(s)
Please list the resources as a list, for example:
- harness_yaml_config

### Terraform Configuration Files
```hcl
resource "harness_yaml_config" "service" {
  path    = "Setup/Applications/My Application/Services/my-service/Index.yaml"
  app_id  = ****
  content = <<EOF
harnessApiVersion: '1.0'
type: SERVICE
artifactType: AWS_LAMBDA
deploymentType: AWS_LAMBDA
description: My Lambda
EOF
}
```

### Expected Behavior
Resource should have been deleted

### Actual Behavior
Resource was not deleted

### Steps to Reproduce
1. `terraform apply`
2. `terraform destroy`

**Comment by micahlmartin** [comment]
This should be addressed in the latest release 0.1.5.

**Comment by chris-cummings** [comment]
Thanks for the fix!!
